src/components/decision-engine/index.js

```diff
@@ -35,14 +38,42 @@ class DecisionEngine {
   }
 
   _sendBlocks (env, cb) {
+    // split into messges of max 512 * 1024 bytes
```

Comment:
Yes, though that is higher than the multiplex limit you implemented in go. So I'm not sure if you could run into the same issue